ext/oj/stream_writer.c

```diff
@@ -44,13 +44,7 @@ static void stream_writer_write(StreamWriter sw) {
     case STRING_IO:
     case STREAM_IO:
     case FILE_IO: {
-        volatile VALUE rs = rb_str_new(sw->sw.out.buf, size);
-
-        // Oddly enough, when pushing ASCII characters with UTF-8 encoding or
-        // even ASCII-8BIT does not change the output encoding. Pushing any
-        // non-ASCII no matter what the encoding changes the output encoding
-        // to ASCII-8BIT if it the string is not forced to UTF-8 here.
-        rs = oj_encode(rs);
+        volatile VALUE rs = rb_utf8_str_new(sw->sw.out.buf, size);
```

Comment:
I think there should be no problem so that the tests was passed added https://github.com/ohler55/oj/commit/73595e6ccf3b140c4a5df598f35ed5e94392d01c